src/plugin.js

```diff
@@ -255,17 +255,42 @@ const initPlugin = function(player, options) {
 
     content.className = 'vjs-errors-dialog';
     content.id = 'vjs-errors-dialog';
+
+    const errorCode = `<div class="vjs-errors-code"><b>${this.localize('Error Code')}:</b> ${(error.type || error.code)}</div>`;
+    const isTimeoutError = error.code === -2;
+
     dialogContent =
      `<div class="vjs-errors-content-container">
-      <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
-        <div class="vjs-errors-code"><b>${this.localize('Error Code')}:</b> ${(error.type || error.code)}</div>
-        ${details}
+        <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
+        ${isTimeoutError ? '' : errorCode}
+        ${isTimeoutError ? '' : details}
       </div>`;
 
     const closeable = display.closeable(!('dismiss' in error) || error.dismiss);
 
-    // We should get a close button
-    if (closeable) {
+    if (isTimeoutError) {
+      dialogContent +=
+       `<div class="vjs-errors-timeout-button-container">
+         <button>${this.localize('Reload Video')}</button>
+         <button>${this.localize('Dismiss')}</button>
+       </div>`;
+      content.innerHTML = dialogContent;
+      display.fillWith(content);
+
+      display.getChild('closeButton').hide();
+
+      const reloadButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:first-child');
+      const dismissButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:last-child');
+
+      player.on(reloadButton, 'click', function() {
+        player.reset();
+        player.src(player.options_.sources);
```

Comment:
player.options_.sources` won't be populated in all cases, I don't think. Here's a suggestion:
```suggestion
        const source = player.currentSource();
        
        player.reset();
        player.src(source);
```

Reply:
Implemented suggestion.